lib/SPIRV/SPIRVReader.cpp

```diff
@@ -3070,6 +3070,49 @@ void SPIRVToLLVM::transFunctionAttrs(SPIRVFunction *BF, Function *F) {
   });
 }
 
+namespace {
+// One basic block can be a predecessor to another basic block more than
+// once (https://github.com/KhronosGroup/SPIRV-LLVM-Translator/issues/2702).
+// This function checks if created PHI's require a fix wrt. this rule.
+static void validatePhiPredecessors(Function *F) {
+  for (BasicBlock &BB : *F) {
+    bool UniquePreds = true;
+    DenseMap<BasicBlock *, unsigned> PredsCnt;
+    for (BasicBlock *PredBB : predecessors(&BB)) {
+      auto It = PredsCnt.try_emplace(PredBB, 1);
+      if (!It.second) {
+        UniquePreds = false;
+        ++It.first->second;
+      }
+    }
+    if (UniquePreds)
+      continue;
+    // `phi` requires an incoming value per each predecessor instance, even
+    // it's the same basic block that has been already inserted as an incoming
+    // value of the `phi`.
+    for (Instruction &Instr : BB) {
+      PHINode *Phi = dyn_cast<PHINode>(&Instr);
+      if (!Phi)
+        continue;
```

Comment:
We should be able to break out of the loop as soon as we see a non-PHI instruction, right?  Alternatively, could we use `BasicBlock::phis()`?